src/components/shared/specPropEditor/SpecPropAutoComplete.tsx

```diff
@@ -0,0 +1,151 @@
+
+import { useIntl } from 'react-intl';
+
+import { autoCompleteDefaultProps } from 'src/components/incompatibleSchemaChange/shared';
+import SpecPropInvalidSetting from 'src/components/shared/specPropEditor/SpecPropInvalidSetting';
+import { stringifyJSON } from 'src/services/stringify';
+import { useFormStateStore_isActive } from 'src/stores/FormState/hooks';
+
+export default function SpecPropAutoComplete({
+    currentSetting,
+    inputLabelId,
+    invalidSettingsMessageId = 'specPropUpdater.error.message',
+    options,
+    renderOption,
+    scope,
+    setErrorExists,
+    updateDraftedSetting,
+    isOptionEqualToValue,
+}: SpecPropAutoCompleteProps) {
+    const intl = useIntl();
+
+    const [inputValue, setInputValue] = useState('');
+    const [invalidSetting, setInvalidSetting] = useState(false);
+
+    const formActive = useFormStateStore_isActive();
+
+    const selection = useMemo(() => {
+        if (!currentSetting || typeof currentSetting !== 'string') {
+            return null;
+        }
+
+        return options.find((option) => option.val === currentSetting) ?? null;
+    }, [currentSetting, options]);
+
+    useEffect(() => {
+        // No setting at all so we're good
+        if (!currentSetting) {
+            setInputValue('');
+            setInvalidSetting(false);
+            return;
+        }
+
+        // We have a setting but could not find a matching option
+        //  Set a flag to show an error and empty out the input
+        if (!selection) {
+            setInputValue('');
+            setInvalidSetting(true);
+            return;
+        }
+
+        setInputValue(selection.label);
+        setInvalidSetting(false);
+    }, [currentSetting, selection]);
+
+    useEffect(() => {
+        setErrorExists(invalidSetting, scope);
+    }, [invalidSetting, scope, setErrorExists]);
+
+    return (
+        <Stack spacing={1}>
+            {invalidSetting ? (
+                <SpecPropInvalidSetting
+                    currentSetting={currentSetting}
+                    invalidSettingsMessageId={invalidSettingsMessageId}
+                    updateDraftedSetting={updateDraftedSetting}
+                />
+            ) : null}
+
+            <Autocomplete
+                {...autoCompleteDefaultProps}
+                disabled={formActive}
+                getOptionLabel={(option) => option.label}
+                inputValue={inputValue}
+                isOptionEqualToValue={(option, optionValue) => {
+                    // We force an undefined some times when we need to clear out the option
+                    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+                    if (!optionValue) {
+                        return false;
+                    }
+
+                    // Allow custom function to run
+                    if (isOptionEqualToValue) {
+                        return isOptionEqualToValue(option, optionValue);
+                    }
```

Comment:
This was mainly added to handle filtering out aliases in the UI but still showing them as selected in the dropdown.